embedded-redis/src/main/java/com/playtika/test/redis/FileUtils.java

```diff
@@ -1,39 +1,46 @@
 package com.playtika.test.redis;
 
 import lombok.experimental.UtilityClass;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.function.Function;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+
+import static java.lang.System.lineSeparator;
+import static java.util.stream.Collectors.joining;
 
 @UtilityClass
 public class FileUtils {
 
-    public static String getFileContent(String filePath) {
-        try (Stream<String> lines = getFileStream(filePath)) {
-            return lines.collect(Collectors.joining("\n"));
-        } catch (Exception e) {
-            throw new IllegalArgumentException("Failed to load file - " + filePath, e);
-        }
-    }
-
-    public static void resolveTemplate(String fileName, Function<String, String> modifyFunc) throws Exception {
-        String fileTemplateContent = FileUtils.getFileContent(fileName + ".template");
+    public static void resolveTemplate(ResourceLoader resourceLoader, String fileName, Function<String, String> modifyFunc) throws Exception {
+        String fileTemplateContent = getFileContent(resourceLoader, fileName + ".template");
         String modifiedFile = modifyFunc.apply(fileTemplateContent);
-        FileUtils.writeToFileInClassesDir(modifiedFile, fileName);
+        writeToFileInClassesDir(modifiedFile, fileName);
     }
 
     public static void writeToFileInClassesDir(String body, String fileName) throws Exception {
-        Path url = Paths.get("target", "classes", fileName);
+        Path url = Paths.get("target", "test-classes", fileName);
```

Comment:
Just a thought, in my use case I do not have target folder but have different one, do you think it can be ClassLoader resource path instead of target path? Thanks.

Reply:
agree, changed